src/lib.rs

```diff
@@ -51,6 +35,71 @@ const CURATION_CONTRACT: [u8; 20] = hex!("8FE00a685Bcb3B2cc296ff6FfEaB10acA4CE15
 substreams_ethereum::init!();
 
 // -------------------- INITIAL MAPS --------------------
+fn pad_address_and_number(address: &Vec<u8>, number: u64) -> Vec<u8> {
+    // Pad the address with leading zeros to make it 32 bytes
+    let mut padded_address = vec![0; 32 - address.len()];
+    padded_address.extend_from_slice(&address);
+
+    // Convert the number to a byte array
+    let number_bytes = number.to_be_bytes();
+
+    // Pad the number with leading zeros to make it 32 bytes
+    let mut padded_number = vec![0; 32 - number_bytes.len()];
+    padded_number.extend_from_slice(&number_bytes);
+
+    // Concatenate the padded address and padded number
+    let mut result = Vec::new();
+    result.extend_from_slice(&padded_address);
+    result.extend_from_slice(&padded_number);
+    result
+}
+
+pub fn keccak256(data: &Vec<u8>) -> [u8; 32] {
+    let mut hasher = Keccak::v256();
+    let mut out: [u8; 32] = [0; 32];
+    hasher.update(data);
+    hasher.finalize(&mut out);
+    out
+}
+
+#[substreams::handlers::map]
+fn map_sc(blk: eth::Block) -> Result<Changes, Error> {
+    let mut changes = Changes::default();
+    let mut indexer_stakes = vec![];
+
+    for trx in blk.transactions() {
+        for call in trx.calls.iter() {
+            let _call_index = call.index;
+            if call.state_reverted {
+                continue;
+            }
+
+            for log in call.logs.iter() {
+                if let Some(event) = abi::staking::events::StakeDeposited::match_and_decode(&log) {
+                    for storage_change in &call.storage_changes {
+                        if storage_change.address.eq(&STAKING_CONTRACT) {
+                            let key = pad_address_and_number(&event.indexer, 14);
+                            if storage_change.key == keccak256(&key) {
+                                indexer_stakes.push(IndexerStake {
+                                    indexer: event.indexer.clone(),
+                                    stake: BigInt::from_signed_bytes_be(&storage_change.new_value)
```

Comment:
Is it a signed bytes? I'm quite sure they are all uint256, although not sure if we have `from_unsigned_bytes` 🤔  (should work anyways for most numbers, unless we get really close to the edge where the +- sign bit is located)